sanic/request.py

```diff
@@ -347,6 +382,24 @@ def token(self):
 
         return auth_header
 
+    @property
+    def credentials(self) -> Credentials:
+        """Attempt to return the auth header value.
+
+        Covers NoAuth, Basic Auth, Bearer Token, Api Token authentication schemas.
+
+        :return: A named tuple with token or username and password related to request
+        """
+        prefixes = ("Basic", "Bearer", "Token")
+        auth_header = self.headers.getone("authorization", None)
+
+        if auth_header is not None:
+            for prefix in prefixes:
+                if prefix in auth_header:
+                    header_value = auth_header.partition(prefix)[-1].strip()
+                    return Credentials(_auth_type=prefix, token=header_value)
+        return Credentials(_auth_type=None, token=auth_header)
```

Comment:
Please put this in `parse_credentials` function and follow the pattern used in other properties. Look at `forwarded` as an example.

Reply:
I don't really get. It's a refactored `token` [property](https://github.com/SerGeRybakov/sanic/blob/credentials/sanic/request.py#L369-L383). What's wrong?

Or you want me to refactor token property also using DRY?